Zend/zend_ast.c

```diff
@@ -901,6 +901,59 @@ ZEND_API zend_result ZEND_FASTCALL zend_ast_evaluate(zval *result, zend_ast *ast
 			}
 			return SUCCESS;
 		}
+		case ZEND_AST_PROP:
+		{
+			if (UNEXPECTED(zend_ast_evaluate(&op1, ast->child[0], scope) != SUCCESS)) {
+				ret = FAILURE;
+				break;
+			}
+			if (UNEXPECTED(zend_ast_evaluate(&op2, ast->child[1], scope) != SUCCESS)) {
+				zval_ptr_dtor_nogc(&op1);
+				ret = FAILURE;
+				break;
+			}
+
+			convert_to_string(&op2);
+			if (EG(exception)) {
+				zval_ptr_dtor_nogc(&op1);
+				zval_ptr_dtor_nogc(&op2);
+				ret = FAILURE;
+				break;
+			}
+
+			if (Z_TYPE(op1) != IS_OBJECT) {
+				// FIXME: Reuse zend_wrong_property_read()
+				zend_string *tmp_property_name;
+				zend_string *property_name = zval_get_tmp_string(&op2, &tmp_property_name);
+				zend_error(E_WARNING, "Attempt to read property \"%s\" on %s", ZSTR_VAL(property_name), zend_zval_type_name(&op1));
+				zend_tmp_string_release(tmp_property_name);
+
+				zval_ptr_dtor_nogc(&op1);
+				zval_ptr_dtor_nogc(&op2);
+
+				ZVAL_NULL(result);
+				ret = SUCCESS;
+				break;
+			}
+
+			zend_object *zobj = Z_OBJ(op1);
+			zend_string *name = Z_STR(op2);
+			zval *property_result = zend_read_property_ex(scope, zobj, name, 0, result);
+			if (EG(exception)) {
+				zval_ptr_dtor_nogc(&op1);
+				zval_ptr_dtor_nogc(&op2);
```

Comment:
Can move these out of the if.

Reply:
This is actually not possible, because doing that, and thus calling `zval_ptr_dtor_nogc(&op1);` before `ZVAL_COPY` causes the object to be deallocated prematurely.